activesupport/lib/active_support/message_verifier.rb

```diff
@@ -46,15 +46,17 @@ def initialize(secret, options = {})
     def valid_message?(signed_message)
       return if signed_message.nil? || !signed_message.valid_encoding? || signed_message.blank?
 
-      data, digest = signed_message.split("--")
-      data.present? && digest.present? && ActiveSupport::SecurityUtils.secure_compare(digest, generate_digest(data))
+      parts = signed_message.split('.')
+      parts.size == 3 && parts.all?(&:present?) && untampered?(parts.pop, parts.join('.'))
     end
 
-    # Decodes the signed message using the +MessageVerifier+'s secret.
+    # Decodes the signed message using the +MessageVerifier+'s secret. It also decodes
+    # legacy signed messages which are outdated messages generated by +MessageVerifier+
+    # of the previous rails version.
```

Comment:
Let's be explicit and say: "messages generated on Rails 4.2 or earlier." - Otherwise we'd needlessly have to update the docs again on Rails 5.1.